python/ray/dashboard/client/src/pages/dashboard/tune/Tune.tsx

```diff
@@ -0,0 +1,114 @@
+import { Theme } from "@material-ui/core/styles/createMuiTheme";
+import createStyles from "@material-ui/core/styles/createStyles";
+import withStyles, { WithStyles } from "@material-ui/core/styles/withStyles";
+import React from "react";
+import { connect } from "react-redux";
+import { getTuneInfo } from "../../../api";
+import { StoreState } from "../../../store";
+import { dashboardActions } from "../state";
+import Typography from "@material-ui/core/Typography";
+import WarningRoundedIcon from "@material-ui/icons/WarningRounded";
+import Tab from "@material-ui/core/Tab";
+import Tabs from "@material-ui/core/Tabs";
+import TuneTable from "./TuneTable";
+import TuneTensorBoard from "./TuneTensorBoard";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      backgroundColor: theme.palette.background.paper
+    },
+    tabs: {
+      borderBottomColor: theme.palette.divider,
+      borderBottomStyle: "solid",
+      borderBottomWidth: 1
+    },
+    warning: {
+      fontSize: "0.8125rem"
+    },
+    warningIcon: {
+      fontSize: "1.25em",
+      verticalAlign: "text-bottom"
+    }
+  });
+
+const mapStateToProps = (state: StoreState) => ({
+  tuneInfo: state.dashboard.tuneInfo
+});
+
+const mapDispatchToProps = dashboardActions;
+
+class Tune extends React.Component<
+  WithStyles<typeof styles> &
+    ReturnType<typeof mapStateToProps> &
+    typeof mapDispatchToProps
+> {
+  timeout: number = 0;
+
+  state = {
+    tabIndex: 0
+  };
+
+  refreshTuneInfo = async () => {
+    try {
+      const [tuneInfo] = await Promise.all([getTuneInfo()]);
+      this.props.setTuneInfo({ tuneInfo });
+    } catch (error) {
+      this.props.setError(error.toString());
+    } finally {
+      this.timeout = window.setTimeout(this.refreshTuneInfo, 1000);
+    }
+  };
+
+  async componentDidMount() {
+    await this.refreshTuneInfo();
+  }
+
+  async componentWillUnmount() {
+    window.clearTimeout(this.timeout);
+  }
+
+  handleTabChange = (event: React.ChangeEvent<{}>, value: number) => {
+    this.setState(prevState => ({
```

Comment:
Don't need to use a function unless the state update depends on the previous state.